lib/bundle/commands/check.rb

```diff
@@ -5,75 +5,38 @@ module Commands
     module Check
       module_function
 
-      def reset!
-        @dsl = nil
-        Bundle::CaskDumper.reset!
-        Bundle::BrewDumper.reset!
-        Bundle::MacAppStoreDumper.reset!
-        Bundle::TapDumper.reset!
-        Bundle::BrewServices.reset!
-      end
-
-      def run
-        if any_taps_to_tap? ||
-           any_casks_to_install? ||
-           any_apps_to_install? ||
-           any_formulae_to_install? ||
-           any_formulae_to_start?
-          puts "brew bundle can't satisfy your Brewfile's dependencies."
-          puts "Satisfy missing dependencies with `brew bundle install`."
-          exit 1
-        else
-          puts "The Brewfile's dependencies are satisfied."
-        end
-      end
-
-      def any_casks_to_install?
-        @dsl ||= Bundle::Dsl.new(Bundle.brewfile)
-        requested_casks = @dsl.entries.select { |e| e.type == :cask }.map(&:name)
-        requested_casks.any? do |c|
-          !Bundle::CaskInstaller.cask_installed_and_up_to_date?(c)
-        end
-      end
+      ARROW = "→"
+      FAILURE_MESSAGE = "brew bundle can't satisfy your Brewfile's dependencies."
+      SUCCESS_MESSAGE = "The Brewfile's dependencies are satisfied."
 
-      def any_formulae_to_install?
-        @dsl ||= Bundle::Dsl.new(Bundle.brewfile)
-        requested_formulae = @dsl.entries.select { |e| e.type == :brew }.map(&:name)
-        requested_formulae.any? do |f|
-          !Bundle::BrewInstaller.formula_installed_and_up_to_date?(f)
-        end
+      def output_errors?
```

Comment:
Still think these methods would be better off as direct calls to `ARGV.include`

Reply:
I remembered where I've used these that really came in handy: the tests. They already existed before I involved them in tests; they were just hella convenient when I needed them.

```ruby
allow(Bundle::Commands::Check).to receive(:exit_on_first_error?).and_return(true)
```

I kinda remember having some trouble setting up tests with `--verbose` enabled. I remember spending a couple hours on it. Then I realized that I could inject the above and it worked.

If I `:focus` on the test on line 153 of `check_command_spec.rb` and comment out

```ruby
allow(ARGV).to receive(:include?).and_return(true)
```

which seems like it actually _should_ enable verbose mode, too, but it's necessary to get a brewfile loaded. Without it, the test cannot load the brewfile:

```
Run options: include {:focus=>true}
F

Failures:

  1) Bundle::Commands::Check when verbose mode is not enabled stops checking after the first missing formula
     Failure/Error: expect { do_check }.to raise_error(SystemExit)

       expected SystemExit, got #<NoMethodError: undefined method `value' for ["--pattern", "spec/**{,/*/**}/*_spec.rb"]:Array
       Did you mean?  values_at> with backtrace:
         # ./lib/bundle/bundle.rb:51:in `brewfile'
         # ./lib/bundle/checker.rb:67:in `check'
         # ./lib/bundle/commands/check.rb:20:in `run'
         # ./spec/check_command_spec.rb:7:in `do_check'
         # ./spec/check_command_spec.rb:161:in `block (4 levels) in <top (required)>'
         # ./spec/check_command_spec.rb:161:in `block (3 levels) in <top (required)>'
         # ./spec/spec_helper.rb:40:in `block (3 levels) in <top (required)>'
         # /Users/colin/.rvm/gems/ruby-2.3.1/gems/bundler-1.16.1/lib/bundler.rb:297:in `block in with_clean_env'
         # /Users/colin/.rvm/gems/ruby-2.3.1/gems/bundler-1.16.1/lib/bundler.rb:540:in `with_env'
         # /Users/colin/.rvm/gems/ruby-2.3.1/gems/bundler-1.16.1/lib/bundler.rb:297:in `with_clean_env'
         # ./spec/spec_helper.rb:40:in `block (2 levels) in <top (required)>'
     # ./spec/check_command_spec.rb:161:in `block (3 levels) in <top (required)>'
     # ./spec/spec_helper.rb:40:in `block (3 levels) in <top (required)>'
     # /Users/colin/.rvm/gems/ruby-2.3.1/gems/bundler-1.16.1/lib/bundler.rb:297:in `block in with_clean_env'
     # /Users/colin/.rvm/gems/ruby-2.3.1/gems/bundler-1.16.1/lib/bundler.rb:540:in `with_env'
     # /Users/colin/.rvm/gems/ruby-2.3.1/gems/bundler-1.16.1/lib/bundler.rb:297:in `with_clean_env'
     # ./spec/spec_helper.rb:40:in `block (2 levels) in <top (required)>'

Finished in 0.053 seconds (files took 0.28185 seconds to load)
1 example, 1 failure
```

So, at this point, if you can suggest an improvement to the test setup that correctly enables verbose mode, I'll be able to remove those two methods. Otherwise, they basically have to stay in order to setup the test harness.